### DISCUSSION TITLE
What does an AP compatible page need to do for Tusky to open links to it within the app?

### DISCUSSION THREAD

**Discussion by j3j5** [tuskyapp/Tusky#3705] [Q&A] [answered]
Hi, 

I've been playing with my own AP compatible server to post messages into the fediverse. So far, the basic functionality seems to be working fine but I noticed that when I post a link to any of the messages from the accounts within tusky, they don't open as a message on the app, but get the option to open them with a browser. What's my page missing? After looking through the issues, my understanding is that the app performs a search against my logged in instance for that URL and, if it returns something, it opens in-app. On my case, pasting the URL on Tusky's search surfaces the message, but clicking on a link to the same message does not open in-app.

See examples:

This is a message from one of my servers: 
https://bots.uy/BigBo_Energy/5491478923

Searching that URL on Tusky returns the message, see screenshot:
![Screenshot_20230601-170229](https://github.com/tuskyapp/Tusky/assets/1239921/a40d5494-6684-459f-a32d-2e0280d643c9)
4839552

Now, if I post the same url on a message and then click on it, it does not open. What's missing? You can test by clicking on that same URL from the following post:
https://mastodon.uy/@j3j5/110469089468384237

Thanks!

**Comment by nikclayton** [comment, marked as answer]
There's no standard for a URL that Tusky should definitely open, so Tusky checks the URL against some regexps.

The list is in https://github.com/tuskyapp/Tusky/blob/01b3cb3a53b1e08ed26e747ddbddb1e7b891ce9d/app/src/main/java/com/keylesspalace/tusky/util/LinkHelper.kt#L287.

**Reply by j3j5** [reply]
Oh, I see, thanks, I wish I saw this before starting to post, oh well, thanks anyway!
